network-store-model/src/main/java/com/powsybl/network/store/model/TapChangerStepAttributes.java

```diff
@@ -47,5 +50,11 @@ public abstract class AbstractTapChangerStepAttributes {
     private Integer side;
 
     @JsonIgnore
-    public abstract TapChangerType getType();
+    public TapChangerType getType() {
+        if (alpha == null) {
```

Comment:
We should not determine the object's type from its values like this. The type should not change once set.

Reply:
I changed, but anyway no field is final because the class use builder so it can't be immutable.